File: Java/ToyLanguageGUI_exam_final/src/repository/IRepository.java
package repository;

import domain.PrgState;

import java.util.List;

public interface IRepository {
    void addPrgState(PrgState p);
    void logProgStateExec(PrgState p);
    List<PrgState> getPrgList();
    void setPrgList(List<PrgState> l);
}

File: ToyInterpretor/ToyInterpretor/Repository/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyInterpretor.Domain;

namespace ToyInterpretor.Repository
{
    interface IRepository
    {
        void addPrgState(PrgState p);
        PrgState getCurrentPrgState();
        void logProgStateExec();
    }
}
